docs/archive_data_migration_to_berkeley.md

```diff
@@ -0,0 +1,470 @@
+
+1. Install the required `mina-archive-blocks` and `mina-missing-blocks-auditor` scripts that are packed in the `minaprotocol/mina-rosetta:1.4.0-c980ba8-bullseye` Docker image.
+
+2. Export the `BLOCKS_BUCKET:
+
+   ```sh
+   export BLOCKS_BUCKET="https://storage.googleapis.com/my_bucket_with_precomputed_blocks"
+
+3. Run the `mina-missing-blocks-auditor` script from the database host:
+
+   ```sh
+   download-missing-blocks.sh mainnet {db_user} {db_password}
+   ```
+
+##### Bad ledger hashes
+
+To verify Mainnet archive data, the replayer application was developed. You must run the replayer application against your existing Mainnet database to verify the blockchain state.
+
+To run replayer:
+
+```sh
+mina-replayer --archive-uri {db_connection_string} --input-file reference_replayer_input.json --output-file reference_replayer_output.json --checkpoint-interval 100
+```
+
+> :warning: Running a replayer from scratch on a Mainnet database can take up to a couple of days. The recommended best practice is to break the replayer into smaller parts by using the checkpoint capabilities of the replayer.
+
+> :warning: ** You must run replayer from the Mainnet version. You can run it from the Docker image at minaprotocol/mina-archive:1.4.0-c980ba8-bullseye
+
+### Google Cloud bucket with Mainnet precomputed blocks
+
+The Mainnet to Berkeley archive data migration requires access to precomputed blocks that are uploaded by daemons connected to the Mainnet network. The Berkeley migration app uses the gsutil app to download blocks. If you didn't store precomputed blocks during first phase of migration, you can use the precomputed blocks provided by Mina Foundation:
+
+`gs://mina_network_block_data`
+
+The best practice is to collect precomputed blocks by yourself or by other third parties to preserve idea of decentralization. 
+
+## Migration applications
+
+Migration applications are distributed as part of the Docker archive, daemon Dockers, or Debian packages.
+
+You can choose packages as you wish. The following commands help you install Debian packages or Docker images.
+
+### Debian packages
+
+```
+CODENAME=bullseye
+CHANNEL=unstable
+# Berkeley nightly version
+VERSION=2.0.0rampup8-berkeley-b9e1116
+
+echo "deb [trusted=yes] http://packages.o1test.net $CODENAME $CHANNEL" | tee /etc/apt/sources.list.d/mina.list
+apt-get update
+apt-get install --allow-downgrades -y "mina-berkeley=$VERSION" "mina-replayer=$VERSION"
+```
+
+### Docker images
+
+The Berkeley migration app:
+
+```
+docker pull gcr.io/o1labs-192920/mina-berkeley:2.0.0rampup8-berkeley-b9e1116-bullseye-berkeley
+```
+
+Replayer:
+
+```
+docker pull gcr.io/o1labs-192920/mina-archive:2.0.0rampup8-berkeley-b9e1116-bullseye
+```
+
+## Mainnet Genesis Ledger
+
+## Berkeley database schema files
+
+
+## Migration Steps
```

Comment:
we actually have several migration steps
1. Plan: read, get the migration apps, make sure you can log in to all databases, and so on
2. Gather info and download the required assets
3. Run migration phase 1
4. Run migration phase 2
5. Verify migration
6. Rinse, repeat (which steps can be repeated?)